test/functional/options/defaults_spec.lua

```diff
@@ -328,3 +329,64 @@ describe('XDG-based defaults', function()
     end)
   end)
 end)
+
+
+describe('stdpath()', function()
+  before_each(function()
+    clear({env={
+      XDG_CONFIG_HOME='cat',
```

Comment:
@docwhat `stdpaths_get_xdg_var()` may only be tested in a unit tests. This is *functional* test, it tests how all that is functioning together. `stdpaths_get_xdg_var()` may be splitted into multiple functions, replaced with some library function (e.g. if libuv will provide similar functionality), etc with tests gone, `stdpath()` is going to stay, be a part of an API and questioning “how API works in edge cases” is completely valid.